File: Exercicios_diarios/Revisão_intermediaria/Grid Challenge.cpp
#include <bits/stdc++.h>

using namespace std;

string ltrim(const string &);
string rtrim(const string &);

int main()
{
    int casos, size;
    scanf("%d", &casos);

    for (int _ = 0; _ < casos; _++)
    {
        scanf("%d", &size);
        string str[size], result = "YES", aux;

        for (int i = 0; i < casos; i++)
            scanf("%s", &str[i]);

        for (int i = 0; i < size; i++)
        {
            aux = str[i];
            sort(str[i].begin(), str[i].end());
            if (aux != str[i])
            {
                result = "NO";
                break;
            }
        }
        printf("%s", result);
    }
}

File: Exercicios_diarios/Revisão_intermediaria/Grid Challenge.py
import math
import os
import random
import re
import sys

if __name__ == '__main__':
    casos = int(input())
    for _ in range(casos):
        len = int(input())
        strings = [list(sorted(input())) for i in range(len)]

        result = 'YES'
        for str in list(zip(*strings)):
            if(list(str) != sorted(str)):
                result = 'NO'
                break
        print(result)
